addons/web_editor/static/src/js/wysiwyg/wysiwyg_iframe.js

```diff
@@ -173,6 +183,33 @@ Wysiwyg.include({
             return this._super.apply(this, arguments);
         }
     },
+
+    /**
+     * When the editable is inside an iframe, use the iframe element instead
+     * of the editable to check whether it is part of the subtree.
+     *
+     * @override
+     **/
+    _isEditableInSubTree: function (root) {
+        if (this.options.inIframe) {
+            return root.contains(this.$iframe[0]);
+        } else {
+            return this._super.apply(this, arguments);
+        }
+    },
+
+    /**
+     * @override
+     **/
+    _configureToolbar: function () {
+        this._super.apply(this, arguments);
+        if (this.options.inIframe && !this.options.snippets) {
+            this._onScrollIframe = () => {
+                this.odooEditor.updateToolbarPosition();
+            };
+            this.options.document.addEventListener('scroll', this._onScrollIframe, true);
```

Comment:
> So when there is an iframe we have two scroll handlers then ? The normal one and this one ? Is this intended ?

Yes, it is intended to have a handler on both documents.
When there is an iframe, we could have the scroll bars either in some element in the top document (like in mass_mailing), either in the the iframe (like in marketing automation).
![image](https://github.com/odoo/odoo/assets/65516503/c55c9225-766b-4b83-bddd-963ee06dca6a)
![image](https://github.com/odoo/odoo/assets/65516503/675c0c3d-37c2-46e3-a53f-049eac3b8e6e)
Whenever things get scrolled, the toolbar position might become wrong, so  method ` _positionToolbar`  is called on scroll.
To keep the toolbar in the correct position, we just need to know **when** there's a scroll event. That's what the first commit tries to fix.

> I feel like the toolbar position is computed with only one particular element in mind

Yes, the toolbar position is computed based on the editable's document. Which might be the iframe, and in case it is, the ` _positionToolbar` method must be aware of the iframe's position relative to the parent document, which is done via `this.options.getContextFromParentRect`. This PR does not change anything about it, by the way.
The scroll container's dimensions is not used to position the toolbar.
 It is used hide the toolbar when it would overflow navbar and other stuff.